presto-main/src/main/java/io/prestosql/server/protocol/Slug.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.protocol;
+
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+
+public final class Slug
+{
+    public static Slug createNew()
+    {
+        return fromKey(randomUUID().toString().toLowerCase(ENGLISH).replace("-", ""));
+    }
+
+    public static Slug fromKey(String slugKey)
```

Comment:
I don't think this is used, so I would remove it.

Reply:
We will need this to be JSON-serializable when we have separate dispatcher node. Will drop for now.